src/Raven.Server/RavenServerStartup.cs

```diff
@@ -157,23 +159,47 @@ private async Task RequestHandler(HttpContext context)
                 context.Response.StatusCode = (int)HttpStatusCode.OK;
                 context.Response.Headers["Content-Type"] = "application/json; charset=utf-8";
                 context.Response.Headers[Constants.Headers.ServerVersion] = RavenVersionAttribute.Instance.AssemblyVersion;
+                // Check context url AND TW clients
+                var isQueryBool = (IsQuery(context) && TrafficWatchManager.HasRegisteredClients);
```

Comment:
It is better to do this at the `_router` level. Where we'll have an indication on the action itself if we want to record its contents.

Reply:
Actually, would be better if we could just add the data to log directly to the context directly.